src/main/java/mat/dao/clause/impl/MeasureDAOImpl.java

```diff
@@ -232,6 +232,14 @@ public MeasureShareDTO extractDTOFromMeasure(Measure measure) {
         return session.createQuery(query).getResultList();
     }
 
+    @Override
+    public Measure getMeasureByMeasureId(String measureId) {
+        if (isEmpty(measureId)) {
+            return null;
+        }
+        return getSessionFactory().getCurrentSession().get(clazz, measureId);
+    }
+
```

Comment:
Is this similar to `GenericDAO.find(id)`?

Reply:
To return a null when Measure Id is not found in Database.

Reply:
Gotcha, and to avoid a try/catch block.